tests/unit/src/test_conditionAssignments.cpp

```diff
@@ -0,0 +1,125 @@
+/* test_conditionAssignments.cpp
+Copyright (c) 2024 by Peter van der Meer
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "es-test.hpp"
+
+// Include only the tested class's header.
+#include "../../../source/ConditionAssignments.h"
+
+// Include a helper for creating well-formed DataNodes (to enable creating non-empty ConditionSets).
+#include "datanode-factory.h"
+// Include a helper for capturing & asserting on logged output.
+#include "output-capture.hpp"
+
+// Include ConditionStore, to enable usage of them for testing ConditionSets.
+#include "../../../source/ConditionsStore.h"
+
+// ... and any system includes needed for the test file.
+#include <cstdint>
+#include <map>
+#include <string>
+
+namespace { // test namespace
+using Conditions = std::map<std::string, int64_t>;
+// #region mock data
+
+
+
+// #endregion mock data
+
+
+
+// #region unit tests
+SCENARIO( "Creating ConditionAssignments" , "[ConditionAssignments][Creation]" ) {
+	GIVEN( "no arguments" ) {
+		const auto set = ConditionAssignments{};
+		THEN( "no conditions are created" ) {
+			REQUIRE( set.IsEmpty() );
+		}
+	}
+}
+
+SCENARIO( "Extending ConditionAssignments", "[ConditionAssignments][Creation]" ) {
+
+	OutputSink warnings(std::cerr);
+
+	GIVEN( "empty ConditionAssignments" ) {
+		auto set = ConditionAssignments{};
+		REQUIRE( set.IsEmpty() );
+
+		THEN( "no assignments are added from empty nodes" ) {
+			const std::string validationWarning = "Error: Loading empty (sub)condition:\n";
+			set.Load(DataNode{});
```

Comment:
Running the unit tests locally (built with clang) this call results in a vector subscript out of bounds error.
This empty data node has no tokens. When the node is passed to `ConditionSet::Load`, it expects at least one token, so it calls `node.Tokens(0)`, which either fails a debug assertion or results in a segfault, because there isn't actually anything there.
I would suggest that this test case should just be removed: DataFile should never produce a DataNode with no tokens, such an object has no meaning. It real DataNodes can actually be empty, we have a much bigger problem.

Reply:
Thanks for finding this!

I changed it to be a node without any children, because that should also register as empty.